core/src/main/java/com/microsoft/applicationinsights/internal/channel/common/TransmissionPolicyManager.java

```diff
@@ -80,11 +93,45 @@ public void run() {
         }
     }
 
+    /**
+     * Create the {@link TransmissionPolicyManager} and set the ability to throttle.
+     * @param throttlingIsEnabled Set whether the {@link TransmissionPolicyManager} can be throttled.  
+     */
     public TransmissionPolicyManager(boolean throttlingIsEnabled) {
         suspensionDate = null;
         this.throttlingIsEnabled = throttlingIsEnabled;
+        this.transmissionHandlers = new ArrayList<TransmissionHandler>();
+        this.backoffManager = new SenderThreadsBackOffManager(new ExponentialBackOffTimesPolicy());
     }
 
+    /** 
+     * Suspend the transmission thread according to the current back off policy.
+     */
+    public void backoff() {
+    	policyState.setCurrentState(TransmissionPolicy.BACKOFF);
+    	long backOffMillis = backoffManager.backOffCurrentSenderThreadValue();
+        if (backOffMillis > 0)
+        {
+        	long backOffSeconds = backOffMillis / 1000;
+        	InternalLogger.INSTANCE.logAlways(InternalLogger.LoggingLevel.TRACE, "App is throttled, telemetry will be blocked for %s seconds.", backOffSeconds);
```

Comment:
Possibly, I can remove the log always, The backoff is called in large increments so it would become chatty if there were multiple threads and if we were unable to send data for a large amount of time.